sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreConf.scala

```diff
@@ -41,6 +41,13 @@ class StateStoreConf(
   /** Minimum versions a State Store implementation should retain to allow rollbacks */
   val minVersionsToRetain: Int = sqlConf.minBatchesToRetain
 
+  /**
+   * Minimum number of stale snapshot version files that need to be present
```

Comment:
Not necessarily a stale snapshot version because changelog version can also be cleaned up.

_Minimum number of stale checkpoint versions that need to be present in the DFS checkpoint directory for old state checkpoint version deletion to be invoked. This is to amortize the cost of discovering and deleting old checkpoint versions._